File: pset6/cash.py
#!/usr/bin/env python
from cs50 import get_float
import sys
a = 25
b = 10
c = 5
d = 1
coins = 0
# check the validity of input hight
while True:
    pass
    dollar = get_float("Change owed:")
    if dollar > 0:
        break
# convert dollars into pins
pins = round(100 * dollar, 2)
# check if pis is divisble by quarters
if pins % a == 0:
    print(pins/a)
    sys.exit()

while pins > a:
    coins += 1
    pins -= a

while pins >= b:
    coins += 1
    pins -= b

while pins >= c:
    coins += 1
    pins -= c

while pins >= d:
    coins += 1
    pins -= d
# print the minimum numbe number of coins
print(coins)

File: pset1/cash.c
#include<stdio.h>
#include<math.h>
#define a 25
#define b 10
#define c 5
#define d 1
int main()
{
    float dollar;
    int coin = 0, pins;
    do
    {
        printf("Change owed:");
        scanf("%f", &dollar);
    }
    while (dollar <= 0);
    //convert dollars into pins
    pins = round(dollar * 100);
    //check if pis is divisble by quarters
    if (pins % a == 0)
    {
        printf("%d\n", pins / a);
        return 0;
    }
    while (pins > a)
    { 
        coin++;
        pins -= a;	
    }
    while (pins >= b)
    { 
        coin++;
        pins -= b;	
    }
    while (pins >= c)
    { 
        coin++;
        pins -= c;
    }
    while (pins >= d)
    { 
        coin++;
        pins -= d;
    }
    //print the minimum numbe number of coins
    printf("%d\n", coin);	
}
